app.server/query/validateQuery.ts

```diff
@@ -25,5 +25,15 @@ export function validateQuery(o: Query) {
         throw 'Query validation failed. Bbox wrong length.'
     }
 
+    // Set offset (page start) to default of 0 if not present or invalid
```

Comment:
Don't get this at all - seems to not do anything, but potentially causes side-effects modifying the argument?

Reply:
Its checking for an undefined offset or offset < 0 which are both invalid and should be set to 0 if that's the case. Offset is the start of the window for results in this case.

Reply:
likewise with the maximum of 50, don't want to set an undefined maximum as its essentially a quick way to make an unpleasant request with 12k results

Reply:
It's clear what the code chunk in isolation is doing ('set the values to within allowed sensible bounds'). But this function is called `validateQuery` and its contract is to validate the query - in this case by throwing if the query is invalid.  It explicitly says :

    /* Throws an exception if the query isn't good. */
    export function validateQuery(o: Query) {

The new lines do something completely different and unexpected.

Could you think of an alternative way to do what you want? Such that it

- doesn't break the contract of the function 
- doesn't cause surprising mutation / side-effecting of the function's argument?

The existing project attempts to follow the functional style throughout, and functions are supposed to be pure (OK this one is exception-throwing, but ...!) and data structures *at least treated as* immutable (even tho being Javascript, everything is mutable).

The easiest thing to do might be to throw an exception if the values are out of range, as the existing code does - after all why would they be wrong unless bug or malevolence ? Allowing default values is something different and that should happen in model binding phase - if it's really necessary?

If it is necessary to "correct" the query for some reason I don't see here, I think at min this needs to be in its own function (possibly in a different place in the app) that explains what it's actually doing, e.g. 

    correctQueryToHaveAllowedLimitAndPageRange(query)

Reply:
Hey @mattdebont did you have a look at this one?